dali/operators/random/noise/shot_noise.h

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_RANDOM_NOISE_SHOT_NOISE_H_
+#define DALI_OPERATORS_RANDOM_NOISE_SHOT_NOISE_H_
+
+#include "dali/operators/random/rng_base.h"
+#include "dali/pipeline/operator/arg_helper.h"
+#include "dali/operators/random/rng_base_gpu.h"
+#include "dali/operators/random/rng_base_cpu.h"
+
+#define DALI_SHOT_NOISE_TYPES \
+  (uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t)
+
+namespace dali {
+
+template <typename Backend, typename T>
+class shot_noise_impl {
+ public:
+  using Dist =
+      typename std::conditional_t<std::is_same<Backend, GPUBackend>::value,
+                                  curand_poisson_dist,
+                                  std::poisson_distribution<uint32_t>>;
+
+  DALI_HOST_DEV explicit shot_noise_impl(float peak = 12)
+      : peak_{peak} {}
+
+  template <typename Generator>
+  DALI_HOST_DEV inline T operator()(T in_val, Generator& g) {
+    float norm = ConvertNorm<float>(in_val);
+    Dist dist(peak_ * norm);
+    return ConvertSatNorm<T>(dist(g) / peak_);
```

Comment:
Store the reciprocal and multiply here.
```suggestion
    return ConvertSatNorm<T>(dist(g) * ipeak_);
```